lib/client/protobuf_signal_fx_client.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+// Copyright (C) 2015 SignalFx, Inc. All rights reserved.
+
+var winston = require('winston'); // Logging library
+
+var SignalFxClient = require('./signal_fx_client').SignalFxClient;
+var conf = require('./conf');
+
+var protocolBuffers = require('./proto').protocolBuffers;
+
+/**
+ * SignalFx API client data handler that uses Protocol Buffers.
+ * This class presents the interfaces that handle the serialization of data
+ * using Protocol Buffers
+ *
+ * @param apiToken
+ * @constructor
+ */
+function ProtoBufSignalFx(apiToken) {
+  SignalFxClient.apply(this, arguments);
+}
+
+ProtoBufSignalFx.prototype = Object.create(SignalFxClient.prototype);
+ProtoBufSignalFx.prototype.constructor = SignalFxClient;
+
+ProtoBufSignalFx.prototype.getHeaderContentType = function () {
+  return conf.PROTOBUF_HEADER_CONTENT_TYPE;
+};
+
+ProtoBufSignalFx.prototype._addToQueue = function (metricType, datapoint) {
+  var protobufDatapoint = new protocolBuffers.DataPoint();
+
+  // Set value by type
+  protobufDatapoint.value = {};
+  if (typeof datapoint.value === 'string') { // string value
+    protobufDatapoint.value.strValue = datapoint.value;
+  } else if (typeof datapoint.value === 'number' && datapoint.value % 1 !== 0) { // 'double' value
+    protobufDatapoint.value.doubleValue = +datapoint.value;
+  } else if (typeof datapoint.value === 'number' && datapoint.value % 1 === 0) { // int value
+    protobufDatapoint.value.intValue = Math.round(+datapoint.value);
```

Comment:
From previous review: Math.round is unnecessary if datapoint.value % 1 === 0 (which is in the if statement).